src/gordon_gcp/plugins/service/gdns_publisher.py

```diff
@@ -48,6 +50,101 @@
 DNS_CHANGES_DONE = 'done'
 
 
+class GDNSPublisherBuilder:
+    """Build and configure a :class:`GDNSPublisher` object.
+
+    Args:
+        config (dict): Google Cloud DNS API related configuration.
+        success_channel (asyncio.Queue): queue to place a successfully
+            published message to be further handled by the ``gordon``
+            core system.
+        error_channel (asyncio.Queue): queue to place a message met
+            with errors to be further handled by the ``gordon`` core
+            system.
+        kwargs (dict): Additional keyword arguments to pass to the
+            publisher.
+    """
+    def __init__(self, config, success_channel, error_channel, **kwargs):
+        self.config = config
+        self.success_channel = success_channel
+        self.error_channel = error_channel
+        self.kwargs = kwargs
+
+    def _validate_config(self):
+        errors = []
+        # TODO (lynn): keyfile won't be required once we support other
+        #              auth methods
+        if not self.config.get('keyfile'):
+            msg = ('The path to a Service Account JSON keyfile is required to '
+                   'authenticate for Google Cloud DNS.')
+            errors.append(msg)
+
+        if not self.config.get('project'):
+            msg = 'The GCP project where Cloud DNS is located is required.'
+            errors.append(msg)
+
+        if not self.config.get('dns_zone'):
+            msg = 'A dns zone is required to build correct A records.'
+            errors.append(msg)
+        elif not self.config.get('dns_zone', '').endswith('.'):
+            msg = ('A dns zone must be an FQDN and end with the root zone '
+                   '(".").')
+            errors.append(msg)
+
+        if not self.config.get('managed_zone'):
+            msg = ('A managed zone is required to publish records to Google '
+                   'Cloud DNS.')
+            errors.append(msg)
+
+        try:
+            timeout = float(self.config.get('publish_wait_timeout', 60))
```

Comment:
It seems potentially confusing to have a default here.  If you want to do an float like this, maybe use something more obvious it doesn't matter what it is.  0 maybe?

Reply:
Ah, confusing because the default is essentially thrown away? I feel like it might be more confusing if it's 0 here, and [60 here](https://github.com/spotify/gordon-gcp/pull/26/files/5b70925ab427e9695dfc66d82477420e8314b06d#diff-8f53676280c6a88b173ebd83ba0803c0R169). 

I did `float` because timedelta does accept it, and I figured for the potential to want to manage timeout a little more fine-grained. But I'm just as content w/ forcing integers.